tasks/paymaster.ts

```diff
@@ -0,0 +1,98 @@
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { parseEther } from "ethers/lib/utils";
+import { deploymentSalt } from "../src/utils";
+import { Paymaster } from "../typechain/hardhat-diamond-abi/HardhatDiamondABI.sol";
+
+// Tasks.
+
+interface PaymasterDeployParams { }
+
+task("paymaster-deploy", "Deploys the main Paymaster contract").setAction(
+  async (_params: PaymasterDeployParams, hre) => {
+    const { address: marketplaceAddr } = await hre.deployments.get("Marketplace");
+    await deployPaymaster(hre, marketplaceAddr);
+  }
+);
+
+interface PaymasterUpdateFacetsParams { }
+
+task("paymaster-update-facets", "Updates facets of our Paymaster")
+  .setAction(async (_params: PaymasterUpdateFacetsParams, hre) => {
+    const { deployments, getNamedAccounts } = hre;
+    const { deployer } = await getNamedAccounts();
+    // Make sure that the fast is known from our tooling.
+    const { address } = await deployments.get("Paymaster");
+    console.log(`Updating paymaster diamond facets at ${address}...`);
+    await deployments.diamond.deploy("Paymaster", {
+      from: deployer,
+      facets: PAYMASTER_FACETS,
+      deterministicSalt: deploymentSalt(hre),
+      log: true,
+    });
+  });
+
+
+interface PaymasterFundParams { }
+
+task("paymaster-fund", "Funds the Paymaster")
```

Comment:
Good job sticking to the weird naming convention I chose...
My logic was that I wanted tasks groupped by contract. Contract name first, task name after. I know it's weird, but it makes it easier to find what you're looking for...